[bond] build failure on Visual Studio 16.6

**Host Environment**
- OS: Windows 10 2004
- Compiler:Microsoft (R) C/C++ Optimizing Compiler Version 19.26.28805 for x86

**To Reproduce**
`./vcpkg install bond`

**Failure logs** 
[stdout-x86-windows.log](https://github.com/microsoft/vcpkg/files/4678746/stdout-x86-windows.log)
[x86-windows.vcpkg_abi_info.txt](https://github.com/microsoft/vcpkg/files/4678747/x86-windows.vcpkg_abi_info.txt)
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/4678748/config-x86-windows-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4678749/install-x86-windows-dbg-out.log)
[patch-x86-windows-0-err.log](https://github.com/microsoft/vcpkg/files/4678750/patch-x86-windows-0-err.log)

```
Starting package 74/74: bond:x86-windows
Building package bond[core]:x86-windows...
-- Note: bond only supports static library linkage. Building static library.
-- Downloading https://github.com/microsoft/bond/archive/8.1.0.tar.gz...
-- Extracting source D:/vcpkg/downloads/microsoft-bond-8.1.0.tar.gz
-- Applying patch fix-install-path.patch
-- Using source at D:/vcpkg/buildtrees/bond/src/8.1.0-02ef21ae99
-- Downloading https://github.com/microsoft/bond/releases/download/8.1.0/gbc-8.1.0-amd64.zip...
-- Extracting source D:/vcpkg/downloads/gbc-8.1.0-amd64.zip
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: D:/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: D:/vcpkg/buildtrees/bond/x86-windows-dbg
    See logs for more information:
      D:\vcpkg\buildtrees\bond\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/bond/portfile.cmake:47 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)
```

**Additional context**
Bond generates ill-formed code like the following in its 'generated' tree:

```
struct var
        {
            // message
            typedef struct : ::bond::reflection::FieldTemplate<
                8189,
                ::bond::reflection::optional_field_modifier,
                SerializableExceptionBase,
                std::string,
                &SerializableExceptionBase::message,
                &s_message_metadata
            > {}  message;
        };
```

which causes the 16.6 compiler to emit:

```
D:\buildtrees\bond\src\8.1.0-02ef21ae99\cpp\generated\bond/core/bond_reflection.h(36): error C5208: unnamed class used in typedef name cannot declare members other than non-static data members, member enumerations, or member classes
```

Hi @chwarr, I tried to update bond to 9.0.0, but I got some error:
```
(23): fatal error C1189: #error:  This file was generated by an older version of the Bond compiler and is incompatible with your version of the Bond library.
```

In _SOURCE/cpp/gemerated/bond/core/bond_types.h_ line 22:
```
#if BOND_MIN_CODEGEN_VERSION > 0x0b03
#error This file was generated by an older version of the Bond compiler and is incompatible with your version of the Bond library.
#endif
```

In _SOURCE/cpp/inc/bond/core/bond_version.h_ line 11:
```
#define BOND_VERSION 0x0900
#define BOND_MIN_CODEGEN_VERSION 0x0c10
```

--- a/ports/bond/CONTROL
+++ b/ports/bond/CONTROL
@@ -1,5 +1,5 @@
 Source: bond
-Version: 8.1.0-4
+Version: 9.0.0
 Description: Bond is a cross-platform framework for working with schematized data. It supports cross-language de/serialization and powerful generic mechanisms for efficiently manipulating data. Bond is broadly used at Microsoft in high scale services.
 Homepage: https://github.com/Microsoft/bond
 Build-Depends: rapidjson, boost-config, boost-utility, boost-assign

--- a/ports/bond/portfile.cmake
+++ b/ports/bond/portfile.cmake
@@ -1,24 +1,28 @@
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
+set(BOND_VER 9.0.0)
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO microsoft/bond
-    REF  8.1.0
-    SHA512 287a2d299036b57e0576903b1f5372bf8071243ada57153c4bf231cdc660faab1e70c60ddde57ac759d941b74af4ba25d81a5d58e8dbf391032b7b226c4cd18c
+    REF  fe6f582ce4beb65644d9338536066e07d80a0289 #9.0.0
+    SHA512 bf9c7436462fabb451c6a50b662455146a37c1421a6fe22920a5c4c1fa7c0fe727c1d783917fa119cd7092dc120e375a99a8eb84e3fc87c17b54a23befd9abc4
     HEAD_REF master
     PATCHES fix-install-path.patch
 )
 
 if (VCPKG_TARGET_IS_WINDOWS)
     vcpkg_download_distfile(GBC_ARCHIVE
-    URLS "https://github.com/microsoft/bond/releases/download/8.1.0/gbc-8.1.0-amd64.zip"
-    FILENAME "gbc-8.1.0-amd64.zip"
-    SHA512 896c9a78fc714e0ea44c37ed36400ec8e5f52d495a8d81aa80834ff6cd6303c7c94e06129f7b2269416a9e0ffb61423e87406db798fb5be7ff00f14981530089
+    URLS "https://github.com/microsoft/bond/releases/download/${BOND_VER}/gbc-${BOND_VER}-amd64.zip"
+    FILENAME "gbc-${BOND_VER}-amd64.zip"
+    SHA512 f4480a3eb7adedfd3da554ef3cdc64b6e7da5c699bde0ccd86b2dd6a159ccacbb1df2b84b6bc80bc8475f30b904cba98085609e42aad929b2b23258eaff52048
     )
 
+    # Clear the generator to prevent it from updating
+    file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/tools/)
     # Extract the precompiled gbc
     vcpkg_extract_source_archive(${GBC_ARCHIVE} ${CURRENT_BUILDTREES_DIR}/tools/)
-    set(FETCHED_GBC_PATH ${CURRENT_BUILDTREES_DIR}/tools/gbc.exe)
+    set(FETCHED_GBC_PATH ${CURRENT_BUILDTREES_DIR}/tools/gbc-${BOND_VER}-amd64.exe)
 
     if (NOT EXISTS "${FETCHED_GBC_PATH}")
         message(FATAL_ERROR "Fetching GBC failed. Expected '${FETCHED_GBC_PATH}' to exists, but it doesn't.")

